apps/server-asset-sg/src/features/asset-edit/asset-edit.repo.ts

```diff
@@ -320,6 +323,45 @@ export class AssetEditRepo implements Repo<AssetEditDetail, number, AssetEditDat
     }
     return detailResult.right as AssetEditDetail;
   }
+
+  private async validateReferencesOrThrow(data: AssetEditData, id?: number): Promise<void> {
+    // check if any of the siblings are in another workgroup
+    for (const assetYId of data.patch.siblingAssetIds) {
+      const siblingCandidate = await this.prismaService.asset.findUnique({
+        where: { assetId: assetYId },
+        select: { assetId: true, workgroupId: true },
+      });
+      if (siblingCandidate?.workgroupId !== data.patch.workgroupId) {
+        throw new Error('Sibling assets must be in the same workgroup as the edited asset');
```

Comment:
Validation errors can be changed to `HttpException` with `HttpStatus.UNPROCESSABE_ENTITY` (but they shouldn't be thrown by the repo).